### PR TITLE
Disable lux renderer from backend

### PR DESCRIPTION
**Pull request by Elfoniok**

### CODE REVIEW

File: golem/node.py [hunk 1]
```diff
@@ -52,6 +52,9 @@ def __init__(self,  # noqa pylint: disable=too-many-arguments
         # and will prevent the IOCP / kqueue reactors from being installed.
         from twisted.internet import reactor
 
+        if mainnet:
+            DockerManager.MAINNET = True
```
Comment (Krigpl) [inline golem/node.py]: same as `DockerManager.MAINNET = mainnet`

File: golem/task/taskmanager.py [hunk 2]
```diff
@@ -63,11 +63,11 @@ class AlreadyRestartedError(Error):
     def __init__(
             self, node_name, node, keys_auth, listen_address="",
             listen_port=0, root_path="res", use_distributed_resources=True,
-            tasks_dir="tasks", task_persistence=True):
+            tasks_dir="tasks", task_persistence=True, mainnet=False):
         super().__init__()
 
-        self.apps_manager = AppsManager()
-        self.apps_manager.load_apps()
+        self.apps_manager = AppsManager(mainnet)
```
Comment (Krigpl) [inline golem/task/taskmanager.py]: I'm not really familiar with this code but this seems a bit suspicious that `AppsManager` is created both in `Node` and here. Wouldn't it make sense here to receive it from `Node` ? (we wouldn't have to provide `mainnet` flag as well then)

File: apps/appsmanager.py [hunk 3]
```diff
@@ -20,13 +20,19 @@ def __init__(self):
...
     """ Temporary solution for apps detection and management. """
-    def __init__(self):
+    def __init__(self, mainnet):
         self.apps = OrderedDict()
+        self._mainnet = mainnet
+
+    def load_all_apps(self):
+        self._load_apps(REGISTERED_CONFIG_FILE)
+        if not self._mainnet:
+            self._load_apps(os.path.join('apps', 'registered_test.ini'))
```
Comment (Krigpl) [inline apps/appsmanager.py]: make a constant like `REGISTERED_CONFIG_FILE`

File: tests/golem/test_opt_node.py [hunk 4]
```diff
@@ -24,6 +25,7 @@ def setUp(self):
         self.args = ['--datadir', self.path]
 
     def tearDown(self):
+        DockerManager.MAINNET = False
```
Comment (Krigpl) [inline tests/golem/test_opt_node.py]: why is it needed? Seems weird that it's only required in `tearDown` without any changes in other places

File: golem/client.py [hunk 5]
```diff
@@ -98,9 +99,11 @@ def __init__(  # noqa pylint: disable=too-many-arguments
             use_concent: bool = False,
             start_geth: bool = False,
             start_geth_port: Optional[int] = None,
-            geth_address: Optional[str] = None) -> None:
+            geth_address: Optional[str] = None,
+            apps_manager: AppsManager = None) -> None:
 
         self.mainnet = mainnet
+        self.apps_manager = apps_manager
```
Comment (Krigpl) [inline golem/client.py]: you don't need that, do you?

File: golem/client.py [hunk 6]
```diff
@@ -98,9 +99,11 @@ def __init__(  # noqa pylint: disable=too-many-arguments
             use_concent: bool = False,
             start_geth: bool = False,
             start_geth_port: Optional[int] = None,
-            geth_address: Optional[str] = None) -> None:
+            geth_address: Optional[str] = None,
+            apps_manager: AppsManager = None) -> None:
```
Comment (Krigpl) [inline golem/client.py]: This will make a bunch of tests (`test_client`) unhappy

**Review by Krigpl — APPROVED** [review]
lgtm, I think

File: golem/client.py (hunk 6 above)
Comment (Elfoniok) [reply golem/client.py]: Thanks

File: golem/client.py (hunk 5 above)
Comment (Elfoniok) [reply golem/client.py]: I need it to pass in start_network function.

Comment (Krigpl) [reply golem/client.py]: Right, github collapsed ~200 lines between these changes and I didn't the context change, derp.

File: apps/appsmanager.py [hunk 7]
```diff
@@ -20,13 +20,21 @@ def __init__(self):
 
 class AppsManager(object):
     """ Temporary solution for apps detection and management. """
-    def __init__(self):
+    def __init__(self, mainnet):
         self.apps = OrderedDict()
+        self._mainnet = mainnet
+        self.isLoaded = False
```
Comment (Krigpl) [inline apps/appsmanager.py:26]: `snake_case` convention
also I think it'd be better to make it private and just make `load_all_apps` do nothing if it's already loaded

File: golem/task/taskmanager.py [hunk 8]
```diff
@@ -63,13 +63,14 @@ class AlreadyRestartedError(Error):
...
+            tasks_dir="tasks", task_persistence=True,
+            apps_manager=AppsManager(False)):
         super().__init__()
 
-        self.apps_manager = AppsManager()
-        self.apps_manager.load_apps()
-
-        apps = list(self.apps_manager.apps.values())
+        self.apps_manager = apps_manager
+        if not apps_manager.isLoaded:
```
Comment (Krigpl) [inline golem/task/taskmanager.py:71]: on the other hand, do you really need that? At this point the apps have already been loaded
maybe except for the tests, but then it should be fixed in the tests, not like this

**Comment by Elfoniok** [comment]
Closing in favor of which is targeted for develop:
https://github.com/golemfactory/golem/pull/2588/files

**Pull request closed without merging**